### ISSUE TITLE
Record token use for each completion call

### ISSUE THREAD

**Issue by SmittieC** [dimagi/open-chat-studio#8]
And add it to the export

And, just to say a little more on tokens:

- There are input (prompt) tokens and output (completion) tokens. Output is [2x the price of input](https://openai.com/pricing) (for GPT-4)
- We must send the entire chat history every time--including the system prompt--to get the response from the LLM. This means the input tokens will grow linearly as the conversation goes on.
- If we have a safety bot enabled, we will have input and output tokens from that additional call to the safety LLM to account for as well
- More advanced features like function calling, which we are using for timers and the IMCI work, will require additional LLM calls like safety bots

Relevant links:

https://python.langchain.com/docs/modules/model_io/models/llms/token_usage_tracking

https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb

**Comment by snopoke** [comment]
(copied from #105)

I can imagine wanting it at the experiment/bot level and at the team level. I also think we'll want different input and output tokens (at least for OpenAI) as they are charged at a different rate.

**Comment by bderenzi** [comment]
Agreed. I think there'd be a number of different aggregations that could be useful (playground, bots, individual chats, per-user, etc)

**Comment by snopoke** [comment]
Started this here: https://github.com/dimagi/open-chat-studio/pull/511
